zio-config/src/main/scala/zio/config/Config.scala

```diff
@@ -15,6 +15,9 @@ sealed trait Config[A] {
 
   final def xmap[B](to: A => B)(from: B => A): Config[B] = Config.Xmap(self, to, from)
 
+  def xmap2[B, C](that: Config[B])(f: (A, B) => Either[ReadError, C])(g: C => Either[WriteError, (A, B)]): Config[C] =
+    (self <*> that).apply[(A, B)]((a, b) => (a, b), t => Some((t._1, t._2))).xmapEither(b => f(b._1, b._2))(g)
+
   final def <*>[B](f: => Config[B]): ProductBuilder[A, B] =
```

Comment:
Remember to change to `|@|` at some point, so we can save `<*>` for `zip`.

Reply:
For sure. We have a good first issue that tracks this one.